biz.aQute.repository/test/aQute/bnd/deployer/repository/TestLocalIndexGeneration.java

```diff
@@ -143,6 +144,30 @@ public static void testFailToGenerate() throws Exception {
 		reporter.clear();
 	}
 
+	public static void testValidGZipFile() throws Exception {
+		PutResult r = repo.put(new BufferedInputStream(new FileInputStream(
+				"testdata/bundles/name.njbartlett.osgi.emf.minimal-2.6.1.jar")), new RepositoryPlugin.PutOptions());
+		File deployedFile = new File(r.artifact);
+
+		File indexFile = IO.getFile("generated/testoutput/index.xml.gz");
+		assertTrue(indexFile.exists());
+
+		GZIPInputStream gzip = new GZIPInputStream(new FileInputStream(indexFile));
+		assertTrue(gzip.read() > -1);
+	}
```

Comment:
The test case should probably close the file and delete the temp files. Windows hates left open files and Linux has limited file handles.